src/main/java/org/jenkinsci/plugins/github_branch_source/BranchOrTagAgeDiscoveryTrait.java

```diff
@@ -0,0 +1,245 @@
+            return true;
+        }
+
+        /**
+         * Out descriptor.
+         */
+        @Symbol("gitHubBranchOrTagAgeHeadAuthority")
+        @Extension
+        public static class DescriptorImpl extends SCMHeadAuthorityDescriptor {
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getDisplayName() {
+                return Messages.BranchOrTagAgeDiscoveryTrait_authorityDisplayName();
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> originClass) {
+                return SCMHeadOrigin.Default.class.isAssignableFrom(originClass);
+            }
+        }
+    }
+
+    /**
+     * Filter that excludes branches and tags that are older than maxDaysOld.
+     */
+    public static class OnlyHeadCommitsNewerThanSCMHeadFilter extends SCMHeadFilter {
+        
+        /**
+         * The maximum age (in days) of branches or tags.
+         */
+        private int maxDaysOld = 0;
+        
+        /**
+         * Constructor.
+         *
+         * @param maxDaysOld the maximum amount of days a branch or tag can be old.
+         */
+        public OnlyHeadCommitsNewerThanSCMHeadFilter(int maxDaysOld) {
+            this.maxDaysOld = maxDaysOld;
+        }
+        
+        /**
+         * Returns the maximum days a branch or tag can be old.
+         *
+         * @return the maximum days a branch or tag can be old.
+         */
+        public int getMaxDaysOld() {
+            return maxDaysOld;
+        }
+        
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isExcluded(@NonNull SCMSourceRequest request, @NonNull SCMHead head) {
+            long lastModified = 0;
+            try {
+                if (head instanceof BranchSCMHead && request instanceof GitHubSCMSourceRequest) {
+                    for (GHBranch b : ((GitHubSCMSourceRequest) request).getBranches()) {
+                        if (b.getName().equals(head.getName())) {
+                            lastModified = b.getOwner().getCommit(b.getSHA1()).getCommitDate().getTime();
+                        }
+                    }
+                } else if (head instanceof TagSCMHead && request instanceof GitHubSCMSourceRequest) {
+                    lastModified = ((TagSCMHead)head).getTimestamp();
+                }
+
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Could not get commit date of latest commit in branch {0}",
+                        new Object[]{head.getName()}
+                        );
+                return false;
+            }
+
+            long daysOld = TimeUnit.MILLISECONDS.toDays(new Date().getTime() - lastModified);
+            if (lastModified != 0 && 
+                    maxDaysOld != 0 && 
```

Comment:
If user puts in a negative number, we warn but don't stop them - treat it as 0.
```suggestion
                    maxDaysOld <= 0 && 
```